xdsl/transforms/convert_linalg_to_loops.py

```diff
@@ -0,0 +1,152 @@
+    IndexType,
+    IntegerAttr,
+    ModuleOp,
+)
+from xdsl.ir import Block, BlockArgument, MLContext, Operation, Region, SSAValue
+from xdsl.ir.affine import AffineDimExpr, AffineMap
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+def indices_for_map(
+    rewriter: PatternRewriter,
+    target_op: Operation,
+    affine_map: AffineMap,
+    input_index_vals: Sequence[SSAValue],
+) -> Sequence[SSAValue]:
+    if affine_map.num_symbols:
+        raise NotImplementedError("Cannot create indices for affine map with symbols")
+    output_indices: list[SSAValue] = []
+    for expr in affine_map.results:
+        if isinstance(expr, AffineDimExpr):
+            output_indices.append(input_index_vals[expr.position])
+        else:
+            used_dims = expr.used_dims()
+            new_index_vals = input_index_vals
+            new_affine_map = AffineMap(
+                affine_map.num_dims, affine_map.num_symbols, (expr,)
+            )
+            if len(used_dims) != affine_map.num_dims:
+                # Remove unused dims
+                selectors = tuple(
+                    dim in used_dims for dim in range(affine_map.num_dims)
+                )
+                new_index_vals = tuple(compress(new_index_vals, selectors))
+                new_affine_map = new_affine_map.compress_dims(selectors)
+
+            rewriter.insert_op_before(
+                apply_op := affine.ApplyOp(
+                    new_index_vals,
+                    AffineMapAttr(new_affine_map),
+                ),
+                target_op,
+            )
+
+            output_indices.append(apply_op.result)
+
+    return output_indices
+
+
+class LowerGenericOpPattern(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: linalg.Generic, rewriter: PatternRewriter) -> None:
+        if op.res:
+            raise NotImplementedError(
+                "lowering for linalg.generic with results not yet supported"
+            )
+
+        # Create loop nest
+
+        ubs = op.get_static_loop_ranges()
+
+        bound_constant_ops = tuple(
+            arith.Constant(IntegerAttr.from_index_int_value(ub)) for ub in ubs
+        )
+        rewriter.insert_op_before_matched_op(bound_constant_ops)
+        bound_constant_values = tuple(op.result for op in bound_constant_ops)
+
+        zero_op = arith.Constant(IntegerAttr.from_index_int_value(0))
+        one_op = arith.Constant(IntegerAttr.from_index_int_value(1))
+        zero_val = zero_op.result
+        one_val = one_op.result
+        if bound_constant_values:
+            rewriter.insert_op_before_matched_op((zero_op, one_op))
+
+        index = IndexType()
+
+        # Insert loop nest
+
+        loop_args: list[BlockArgument] = []
+        insertion_target: Operation = op
+
+        for ub in bound_constant_values:
```

Comment:
Can you document in which direction the loop nest is built? It seems outermost to innermost to me